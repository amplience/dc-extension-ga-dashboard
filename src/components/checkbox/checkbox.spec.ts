import { render } from '@testing-library/svelte';
import Checkbox from './checkbox.svelte';

describe('Checkbox', () => {
  it('should render a checkbox not checked - false supplied', async () => {
    const { container } = render(Checkbox, {
      className: 'test-class',
      checked: false,
      label: 'test label',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a checkbox not checked - undefined checked value', async () => {
    const { container } = render(Checkbox, {
      className: 'test-class',
      label: 'test label',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a checkbox checked', async () => {
    const { container } = render(Checkbox, {
      className: 'test-class',
      checked: true,
      label: 'test label',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a checkbox checked and disabled', async () => {
    const { container } = render(Checkbox, {
      className: 'test-class',
      checked: true,
      disabled: true,
      label: 'test label',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a checkbox unchecked and disabled', async () => {
    const { container } = render(Checkbox, {
      className: 'test-class',
      checked: false,
      disabled: true,
      label: 'test label',
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
