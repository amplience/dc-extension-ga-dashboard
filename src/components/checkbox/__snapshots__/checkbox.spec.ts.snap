// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render a checkbox checked 1`] = `
<div>
  <div
    class="smui-checkbox test-class svelte-1929qnl"
  >
    <div
      class="mdc-checkbox checkbox mdc-checkbox--upgraded"
    >
      <input
        class="mdc-checkbox__native-control "
        type="checkbox"
        value=""
      />
       
      <div
        class="mdc-checkbox__background"
      >
        <svg
          class="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            class="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
         
        <div
          class="mdc-checkbox__mixedmark"
        />
      </div>
    </div>
     
    <label
      class="svelte-1929qnl"
    >
      test label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render a checkbox checked and disabled 1`] = `
<div>
  <div
    class="smui-checkbox test-class svelte-1929qnl"
  >
    <div
      class="mdc-checkbox checkbox mdc-checkbox--disabled mdc-checkbox--upgraded"
    >
      <input
        class="mdc-checkbox__native-control "
        disabled=""
        type="checkbox"
        value=""
      />
       
      <div
        class="mdc-checkbox__background"
      >
        <svg
          class="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            class="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
         
        <div
          class="mdc-checkbox__mixedmark"
        />
      </div>
    </div>
     
    <label
      class="svelte-1929qnl disabled"
    >
      test label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render a checkbox not checked - false supplied 1`] = `
<div>
  <div
    class="smui-checkbox test-class svelte-1929qnl"
  >
    <div
      class="mdc-checkbox checkbox mdc-checkbox--upgraded"
    >
      <input
        class="mdc-checkbox__native-control "
        type="checkbox"
        value=""
      />
       
      <div
        class="mdc-checkbox__background"
      >
        <svg
          class="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            class="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
         
        <div
          class="mdc-checkbox__mixedmark"
        />
      </div>
    </div>
     
    <label
      class="svelte-1929qnl"
    >
      test label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render a checkbox not checked - undefined checked value 1`] = `
<div>
  <div
    class="smui-checkbox test-class svelte-1929qnl"
  >
    <div
      class="mdc-checkbox checkbox mdc-checkbox--upgraded"
    >
      <input
        class="mdc-checkbox__native-control "
        type="checkbox"
        value=""
      />
       
      <div
        class="mdc-checkbox__background"
      >
        <svg
          class="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            class="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
         
        <div
          class="mdc-checkbox__mixedmark"
        />
      </div>
    </div>
     
    <label
      class="svelte-1929qnl"
    >
      test label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render a checkbox unchecked and disabled 1`] = `
<div>
  <div
    class="smui-checkbox test-class svelte-1929qnl"
  >
    <div
      class="mdc-checkbox checkbox mdc-checkbox--disabled mdc-checkbox--upgraded"
    >
      <input
        class="mdc-checkbox__native-control "
        disabled=""
        type="checkbox"
        value=""
      />
       
      <div
        class="mdc-checkbox__background"
      >
        <svg
          class="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            class="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
         
        <div
          class="mdc-checkbox__mixedmark"
        />
      </div>
    </div>
     
    <label
      class="svelte-1929qnl disabled"
    >
      test label
    </label>
  </div>
</div>
`;
